File: app/src/main/java/com/example/estres2/ui/Graficas/GraficasFragment.java
package com.example.estres2.ui.Graficas;

import android.annotation.SuppressLint;
import android.graphics.Color;
import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.estres2.R;
import com.jjoe64.graphview.GraphView;
import com.jjoe64.graphview.GridLabelRenderer;
import com.jjoe64.graphview.series.DataPoint;
import com.jjoe64.graphview.series.LineGraphSeries;

public class GraficasFragment extends Fragment {
    private TextView Graficas;
    private LineGraphSeries<DataPoint> FC1, GSR1;

    public GraficasFragment() {
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @SuppressLint("SetTextI18n")
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View root = inflater.inflate(R.layout.fragment_graficas, container, false);

        Graficas = (TextView) root.findViewById(R.id.text_graficas);
        Graficas.setText("Vista de las gráficas");

        double x, y, y2;
        x = 0;

        GraphView grafica = (GraphView) root.findViewById(R.id.FC);
        GraphView grafica2 = (GraphView) root.findViewById(R.id.GSR);
        GraphView grafica3 = (GraphView) root.findViewById(R.id.FCYGSR);

        GridLabelRenderer gridLabelFC = grafica.getGridLabelRenderer();
        gridLabelFC.setHorizontalAxisTitle("Tiempo (s)");
        gridLabelFC.setVerticalAxisTitle("Frecuencia Cardiaca");

        GridLabelRenderer gridLabelGSR = grafica2.getGridLabelRenderer();
        gridLabelGSR.setHorizontalAxisTitle("Tiempo (s)");
        gridLabelGSR.setVerticalAxisTitle("Kilo Ohm");

        GridLabelRenderer gridLabelFCYGSR = grafica3.getGridLabelRenderer();
        gridLabelFCYGSR.setHorizontalAxisTitle("Tiempo (s)");
        gridLabelFCYGSR.setVerticalAxisTitle("Frecuencia Cardiaca y Kilo Ohm");

        FC1 = new LineGraphSeries<>();
        GSR1 = new LineGraphSeries<>();

        int NumeroPuntos = 500;

        for (int i = 0; i < NumeroPuntos; i++) {
            y = Math.sin(x);
            y2 = Math.cos(x);

            FC1.appendData(new DataPoint(x, y), true, 250);
            GSR1.appendData(new DataPoint(x, y2), true, 250);

            x = x + 0.1;
        }
        grafica.addSeries(FC1);
        grafica.setTitle("Frecuencia Cardiaca");

        grafica2.addSeries(GSR1);
        grafica2.setTitle("Respuesta Galvánica de la Piel");

        grafica3.addSeries(FC1);
        grafica3.addSeries(GSR1);
        grafica3.setTitle("Frecuencia Cardiaca y Respuesta Galvánica de la Piel");

        FC1.setColor(Color.rgb(226, 91, 34));
        FC1.setThickness(6);
        FC1.setDrawBackground(true);
        FC1.setBackgroundColor(Color.argb(60, 95, 226, 156));

        grafica.getViewport().setScalable(true);
        grafica.getViewport().setScrollable(true);
        grafica.getViewport().setScrollableY(true);

        return root;
    }

}

File: app/src/main/java/com/example/estres2/ui/graficas/GraficasFragment.kt
package com.example.estres2.ui.graficas

import android.graphics.Color
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.jjoe64.graphview.series.DataPoint
import com.jjoe64.graphview.series.LineGraphSeries
import com.example.estres2.databinding.FragmentGraphsBinding
import com.jjoe64.graphview.GridLabelRenderer
import kotlin.math.cos
import kotlin.math.sin

class GraficasFragment : Fragment() {
    private var _binding: FragmentGraphsBinding? = null
    private val binding get() = _binding!!
    private lateinit var gridLabelRendererFC: GridLabelRenderer
    private lateinit var gridLabelRendererGSR: GridLabelRenderer
    private lateinit var gridLabelRendererFCYGSR: GridLabelRenderer
    private lateinit var fc1: LineGraphSeries<DataPoint>
    private lateinit var gsr1: LineGraphSeries<DataPoint>

    override fun onCreateView(
            inflater: LayoutInflater,
            container: ViewGroup?,
            savedInstanceState: Bundle?,
    ): View {
        _binding = FragmentGraphsBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initializeObjects()
    }

    override fun onDestroy() {
        super.onDestroy()
        _binding = null
    }

    private fun initializeObjects() {
        var x = 0.0
        var y: Double
        var y2: Double

        fc1 = LineGraphSeries()
        gsr1 = LineGraphSeries()
        fc1.color = Color.rgb(226, 91, 34)
        fc1.thickness = 6
        fc1.isDrawBackground = true
        fc1.backgroundColor = Color.argb(60, 95, 226, 156)

        val pointsNumber = 500
        for (i in 0 until pointsNumber) {
            y = sin(x)
            y2 = cos(x)
            fc1.appendData(DataPoint(x, y), true, 250)
            gsr1.appendData(DataPoint(x, y2), true, 250)
            x += 0.1
        }
        binding.apply {
            gridLabelRendererFC = FC.gridLabelRenderer
            gridLabelRendererFC.horizontalAxisTitle = "Tiempo (s)"
            gridLabelRendererFC.verticalAxisTitle = "Frecuencia Cardiaca"
            gridLabelRendererGSR = GSR.gridLabelRenderer
            gridLabelRendererGSR.horizontalAxisTitle = "Tiempo (s)"
            gridLabelRendererGSR.verticalAxisTitle = "Kilo Ohm"
            gridLabelRendererFCYGSR = FCYGSR.gridLabelRenderer
            gridLabelRendererFCYGSR.horizontalAxisTitle = "Tiempo (s)"
            gridLabelRendererFCYGSR.verticalAxisTitle = "Frecuencia Cardiaca y Kilo Ohm"

            FC.apply {
                addSeries(fc1)
                title = "Frecuencia Cardiaca"
                viewport.isScalable = true
                viewport.isScrollable = true
                viewport.setScalableY(true)
            }

            GSR.apply {
                addSeries(gsr1)
                title = "Respuesta Galvánica de la Piel"
            }

            FCYGSR.apply {
                addSeries(fc1)
                addSeries(gsr1)
                title = "Frecuencia Cardiaca y Respuesta Galvánica de la Piel"
            }
        }
    }
}